src/main/java/com/sorcix/sirc/IrcParser.java

```diff
@@ -51,6 +54,9 @@ protected void parseCommand(final IrcConnection irc, final IrcPacket line) {
 			if (line.isCtcp()) {
 				// reply to CTCP commands
 				if (line.getMessage().startsWith("ACTION ")) {
+                    for (MessageEventListener l : irc.getMessageEventListeners()) {
```

Comment:
All of these for each loops will throw a ConcurrentModificationException if a listener is added or removed while iterating over the list.  The most common case of this could be a listener which removes itself.

Possible solutions:
* Have get*Listeners() copy the structures into a new ArrayList
* When adding or removing a listener perform a copy of the current listener structure
* You could use the ListenerHelper from my threadly project which I PR'ed earlier